src/components/Filter/Filter.jsx

```diff
@@ -1,28 +1,83 @@
-import { useVideo } from "../../contexts";
+import { useNavigate } from "react-router-dom";
+import { useFilter, useVideo } from "../../contexts";
 import "./Filter.css";
 
 export const Filter = () => {
+	const { byLatest, category, filterDispatch } = useFilter();
 	const {
 		videoState: { categories },
 	} = useVideo();
+	const navigate = useNavigate();
+
 	return (
 		<form className="filter_section" onSubmit={(e) => e.preventDefault()}>
 			<div className="filter_container">
-				<button className="filter_button btn--outline--primary">All</button>
+				<button
+					onClick={() => {
+						filterDispatch({
+							type: "CATEGORY",
+							payload: "All",
```

Comment:
Can you move the logic out of html into js function

Reply:
Yeah will do that.